src/java/org/apache/cassandra/db/rows/UnfilteredRowIteratorWithLowerBound.java

```diff
@@ -83,25 +84,23 @@ public UnfilteredRowIteratorWithLowerBound(DecoratedKey partitionKey,
 
     public Unfiltered lowerBound()
     {
-        if (lowerBound != null)
-            return makeBound(lowerBound);
+        if (lowerBoundMarker != null)
+            return lowerBoundMarker;
 
         // The partition index lower bound is more accurate than the sstable metadata lower bound but it is only
         // present if the iterator has already been initialized, which we only do when there are tombstones since in
         // this case we cannot use the sstable metadata clustering values
-        ClusteringBound<?> ret = getPartitionIndexLowerBound();
-        return ret != null ? makeBound(ret) : makeBound(getMetadataLowerBound());
+        lowerBound = getMetadataLowerBound();
```

Comment:
The original idea of `getPartitionIndexLowerBound` was to use the cached index entry if it's available, and to go to the metadata bound if it's not, in neither case going through `maybeInit`. The way it's currently written doesn't make much sense, and I don't think it's doing anything useful, so removing it is a good idea. It was also removed from DSE 6 because the trie indexes do not use a key cache.

However, OSS Cassandra should still benefit from this path, so if you want to preserve what it originally tried to do (I do not insist on doing so), you can rename it to `getKeyCacheLowerBound` and make sure it goes directly to checking `getCachedPosition`.